### ISSUE TITLE
Right clicking on custom device root item throws error 1162

### ISSUE THREAD

**Issue by droohy** [ni/niveristand-instrument-addon-custom-device#145]
We're using the 2020 build of this custom device in VS 2020 R2.  When we click on any Instrument Addon custom device instance in the system explorer in any VS 2020 project, we get error 1162, so we're unable to add any messages.  This does not happen with any other custom device.  We haven't testing in other versions or on other PCs.  Is this reproducible on others' PCs?  Below is an image of the error in a brand new project:

![image](https://user-images.githubusercontent.com/47541764/96500956-aa925380-121d-11eb-9826-49210e1221ef.png)

**Comment by bogdanp-ni** [comment]
@droohy Thank you for reporting this issue, we are currently looking at it. 

I understood that you built your own Instrument Addon, is that correct? Which branch did you use? I'm asking because we're actively developing in the main branch, which means that code from that branch can be less stable than code from release/20.3 branch, which was already released. And if you don't require any code modifications in Instrument Addon, then I suggest you grab the latest release package from here: https://github.com/ni/niveristand-instrument-addon-custom-device/releases/tag/v20.3.0, which is compatible with VS 2020 R2 and does not exhibit the behavior you noticed:

![image](https://user-images.githubusercontent.com/61490253/96574107-b3476000-12d7-11eb-902d-02bbb5fb558d.png)

**Comment by droohy** [comment]
@bogdanp-ni Sorry I wasn't clear in my original post.  The "2020 build of this custom device" I referred to in my original post _is_ the 20.3 release from the github page you linked.  We downloaded and unzipped "Instrument_Addon_2020.zip" and installed the package.  I have already tried uninstalling and reinstalling the package with a fresh download of the .zip file, but that did not resolve the problem.  At first I thought it was maybe an upgrade issue related to our specific sys def since we had used a previous version of VS and this custom device built for us by Wineman, but the issue still happens in brand new sys defs made in VS 2020 R2 as in my screenshot above.

When you tried to reproduce, did you use VS 2020 R2, and did you install the package downloaded from the releases page?

Is there a particular file responsible for right-click plugin functionality I could upload or provide the checksum for or something to confirm that it is in the right state?  Since this issue is CD-specific (other CDs' right click menus work fine), I have to think that it's not a broader VS installation problem, but I don't know enough about the process to be sure.

**Comment by bogdanp-ni** [comment]
@droohy This is rather weird. Yes, for reproducing it, I used a clean machine with VS 2020 R2, downloaded the "Instrument_Addon_2020.zip" from GitHub (from releases page, 20.3.0) and installed the nipkg. And it worked for me, I tested a few hours ago.

Was this a clean machine that you are using? Did you have a previously-installed Instrument Addon on this machine?

The right-click behavior is defined in "C:\Users\Public\Documents\National Instruments\NI VeriStand 2020\Custom Devices\Instrument Addon\Custom Device Instrument Addon.xml". If possible, I would like to take a look at it.

**Comment by droohy** [comment]
@bogdanp-ni 
**Was this a clean machine that you are using? Did you have a previously-installed Instrument Addon on this machine?**
I'm helping support another team on this (I worked at NI for 5 years, so they generally escalate issues through me), so I don't know the full history of the PC.  I know the PC was acquired from our IT department with a fresh Windows install for the purposes of evaluating a 2020 stack of NI software, but I don't know if the team tried to install previous versions of the Instrument Addon.  If that could be an issue, I would definitely assume that it could have happened.

**The right-click behavior is defined in "C:\Users\Public\Documents\National Instruments\NI VeriStand 2020\Custom Devices\Instrument Addon\Custom Device Instrument Addon.xml". If possible, I would like to take a look at it.**
I've gone ahead and zipped up the whole Instrument Addon folder in case you want to check any other contents.
[Instrument Addon.zip](https://github.com/ni/niveristand-instrument-addon-custom-device/files/5409205/Instrument.Addon.zip)

Thanks!
David

**Comment by bogdanp-ni** [comment]
@droohy 
I replaced my Instrument Addon installation with the one you sent me, but I'm still unable to reproduce the issue. Comparing the Custom Device Instrument Addon.xml you sent with the one I had showed that the two files are identical. 

Is this issue showing up consistently for you? Would you be able to test it on another machine?

In the meantime, I am continuing the investigation and will keep you up-to-date with my findings.

**Comment by droohy** [comment]
@bogdanp-ni 
**Is this issue showing up consistently for you?**
Yes.  It does not work in any system definition on the test PC.  According to the test engineers using it, it has never worked in VeriStand 2020 R2.  (It worked in earlier VeriStand versions.)  It still does not work.

**Would you be able to test it on another machine?**
Yes, I just downloaded VeriStand 2020 R2 and the 20.3 release of the Instrument Addon custom device on my machine, and the issue did not replicate.

Would you recommend reinstallation of a particular component (other than the CD itself, which I've already tried) on the test machine that exhibits the problem?  Given that the issue only appears with this custom device and no other, I'm reluctant to point to a VeriStand installation issue, but, again, I don't know enough about where this error is being thrown and why to claim that it could be a VeriStand problem.

**Comment by bogdanp-ni** [comment]
> **Would you be able to test it on another machine?**
> Yes, I just downloaded VeriStand 2020 R2 and the 20.3 release of the Instrument Addon custom device on my machine, and the issue did not replicate.

We've been trying to replicate the problem, as well, but to no avail. We've also tried with various other versions of both VS 2020 and Instrument Addon builds, but it seemed to be just fine.

Considering that this issue is visible only on that one test machine, I cannot recommend a particular component be reinstalled/uninstalled. If possible, I would go the route of clean Windows install on that machine, then adding VS 2020 and Instrument Addon.

**Comment by bogdanp-ni** [comment]
@droohy 

Please let me know if you're ok with me closing this issue.

**Comment by droohy** [comment]
@bogdanp-ni Yes, you can close it.  As a datapoint, I did a repair install of VeriStand 2020 R2 and included repairing all suggested dependencies that NI Package Manager recommended, and after restarting, the same issue occurred.

Either way, the issue seems isolated to a single PC.  Neither of us can reproduce it on other machines, so I don't what else there'd be to explore.

The sys def we're using for test is stable with the hand-mods I made to the XML to add the channels we needed, and the operators have other PCs they can use to modify the sys def if necessary.

Thanks!
David
